tests/interfaces/records/handlers/records-query.spec.ts

```diff
@@ -449,11 +449,44 @@ describe('RecordsQueryHandler.handle()', () => {
       expect(publishedDescendingQueryReply.entries?.[2].descriptor['datePublished']).to.equal(write1Data.message.descriptor.datePublished);
     });
 
-    it('should only return published records and unpublished records that is meant for author', async () => {
-      // write three records into Alice's DB:
+    it('should allow an anonymous unauthenticated query to return published records', async () => {
+      // write 2 records into Alice's DB:
       // 1st is unpublished
-      // 2nd is also unpublished but is meant for (has recipient as) Bob
-      // 3rd is also unpublished but is authored (sent) by Bob
+      // 2nd is published
+      const alice = await DidKeyResolver.generate();
+      const record1Data = await TestDataGenerator.generateRecordsWrite(
+        { author: alice, schema: 'https://schema1', published: false }
+      );
+      const record2Data = await TestDataGenerator.generateRecordsWrite(
+        { author: alice, schema: 'https://schema2', published: true }
+      );
+
+      const recordsWrite1Reply = await dwn.processMessage(alice.did, record1Data.message, record1Data.dataStream);
+      expect(recordsWrite1Reply.status.code).to.equal(202);
+      const recordsWrite2Reply = await dwn.processMessage(alice.did, record2Data.message, record2Data.dataStream);
+      expect(recordsWrite2Reply.status.code).to.equal(202);
+
+      // test correctness for anonymous query
+      const anonymousQueryMessageData = await TestDataGenerator.generateRecordsQuery({
+        filter: { dateCreated: { from: '2000-01-01T10:20:30.123456Z' } }
+      });
```

Comment:
Do you mind adding an option to `generateRecordsQuery` that will make it not generate a persona? I prefer that to `delete`ing `authorization`.

```suggestion
      const anonymousQueryMessageData = await TestDataGenerator.generateRecordsQuery({
        filter: { dateCreated: { from: '2000-01-01T10:20:30.123456Z' } },
        anonymous: true
      });
```

Reply:
Sure, added. Didn't think was worth it due to some minor cascading changes.

Reply:
Sure, added. Didn't think was worth the additional code and cascading effects.